network-traversal-video-call/ACS-TURN-Javascript/index.html

```diff
@@ -0,0 +1,236 @@
+
+    var VideoChat = {
+        // Initialize our connection to the WebSocket.
+        socket: io(),
+      
+        // Call to getUserMedia (provided by adapter.js for cross browser compatibility)
+        // asking for access to both the video and audio streams. If the request is
+        // accepted callback to the onMediaStream function, otherwise callback to the
+        // noMediaStream function.
+        requestMediaStream: function(event){
+            navigator.mediaDevices.getUserMedia({video: true, audio: true})
+            .then(VideoChat.onMediaStream)
+            .catch(function(err) {
+                console.log(err.name + ": " + err.message);
+            });
+        },
+      
+        // The onMediaStream function receives the media stream as an argument.
+        onMediaStream: function(stream){
+          console.log("\nOnMedia\n");
+          // Get the video element.
+          VideoChat.localVideo = document.getElementById('local-video');
+          // Turn the volume down to 0 to avoid echoes.
+          VideoChat.localVideo.volume = 0;
+          VideoChat.localStream = stream;
+          console.log("\n Set attribute\n");
+          // Turn the media stream into a URL that can be used by the video and add it
+          // as the video's `src`. As the video has the `autoplay` attribute it will
+          // start to stream immediately.
+          VideoChat.localVideo.srcObject = stream;
+          // Now we're ready to join the chat room.
+          VideoChat.socket.emit('join', 'test');
+          console.log("\nready to call\n");
+          VideoChat.socket.on('ready', VideoChat.readyToCall);
+          console.log("\nOffer\n");
+          VideoChat.socket.on('offer', VideoChat.onOffer);
+          console.log("\nOnMedia end\n");
+        },
+      
+        // There's not much to do in this demo if there is no media stream. So
+        // let's just stop.
+        noMediaStream: function(){
+          console.log("No media stream for us.");
+          // Sad trombone.
+        },
+      
+        // When we are ready to call, enable the Call button.
+        readyToCall: function(event){
+          console.log("\nReady to call\n");
+        },
+      
+        // Set up a callback to run when we have the ephemeral token to use Twilio's
+        // TURN server.
+        startCall: function(event){
+          console.log("\nStartCall\n");
+          VideoChat.socket.on('token', VideoChat.onToken(VideoChat.createOffer));
+          VideoChat.socket.emit('token');
+        },
+      
+        // When we receive the ephemeral token back from the server.
+        onToken: function(callback){
+          return function(token) {       
+
+            fetch(`getcredentials`)
+                .then(response => response.json())
+                .then(iceServers => {
+                  // Set up a new RTCPeerConnection using the token's iceServers.
+                  VideoChat.peerConnection = new RTCPeerConnection({
```

Comment:
Creating a new iceServers list isn't needed - especially since you're hardcoding the URLs value here
```js
VideoChat.peerConnection = new RTCPeerConnection({
                  iceServers: [
                    { 
                      'urls': 'turn:worldaz.turn.skype.com:3478', 
                      'username': iceServers[0].username, 
                      'credential': iceServers[0].credential
                    }
                  ]
            });
```
```
VideoChat.peerConnection = new RTCPeerConnection({iceServers});